internal/tools/aws/s3.go

```diff
@@ -103,3 +103,16 @@ func (a *Client) S3EnsureBucketDirectoryDeleted(bucketName, directory string, lo
 
 	return nil
 }
+
+// S3EnsureObjectDeleted is used to ensure that the file is deleted.
+func (a *Client) S3EnsureObjectDeleted(bucketName, path string) error {
+	_, err := a.Service().s3.DeleteObject(&s3.DeleteObjectInput{
+		Bucket: aws.String(bucketName),
+		Key:    aws.String(path),
+	})
+	if err != nil {
```

Comment:
is this error `nil` when the object was previously deleted? I just want to make sure this method does not return an error if you try to use it to delete an object that has already been deleted. 'Ensure' in the name makes it sound like it should not return an error if you try to delete something that has already been deleted.

Reply:
Thanks Ian. I have double checked that and it does not return error if object was already deleted or never existed.